internal/handlers/todo_test.go

```diff
@@ -0,0 +1,87 @@
+package handlers_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/shreyas1307/go-server/internal/handlers"
+	"github.com/shreyas1307/go-server/internal/store/models"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockServerStore struct{}
+
+func (m *MockServerStore) AddTodo(ctx context.Context, todo models.Todo) error {
+	return nil
+}
+
+func (m *MockServerStore) GetTodos(ctx context.Context) (models.TodoList, error) {
+	return models.TodoList{}, nil
+}
+
+func TestAddTodoSuccessful(t *testing.T) {
+	th := &handlers.TodoHandler{
+		Store: &MockServerStore{},
+	}
+	todo := models.Todo{
+		ID:          "1",
+		Title:       "Test",
+		Description: "This is a test todo",
+		Completed:   true,
+	}
+
+	todoMarshal, err := json.Marshal(todo)
+	assert.NoError(t, err)
+
+	reader := bytes.NewReader(todoMarshal)
+	request := httptest.NewRequest(http.MethodPost, "/todo/add", reader)
+	recorder := httptest.NewRecorder()
+
+	th.AddTodo(recorder, request)
+
+	res := recorder.Result()
+	assert.Equal(t, http.StatusOK, res.StatusCode)
+
+	b, err := io.ReadAll(res.Body)
+	assert.NoError(t, err)
+	defer res.Body.Close()
+
+	expextedResponse := map[string]interface{}{
+		"successful": true,
+		"message":    "new todo added",
+	}
+	expectedResponseMarshall, err := json.Marshal(expextedResponse)
```

Comment:
Yeah so to simplify this you can do:

```go
res, err := json.Marshal(map[string]interface{}{"successful": true,"message":    "new todo added"})
```

Reply:
This also highlights your need for a response object - using arbritary maps can lead to problems 😝 (as in server updates or removes a field, and now the test is broken.